src/mux_client_impl/builder.rs

```diff
@@ -0,0 +1,99 @@
+use super::{Error, Session};
+use crate::SessionBuilder;
+
+use std::fs;
+use std::io;
+use std::path::Path;
+use std::process::Stdio;
+use std::str;
+
+use tempfile::Builder;
+
+use tokio::process;
+
+pub(crate) async fn just_connect<S: AsRef<str>>(
```

Comment:
I _think_ the body of this function is almost entirely shared between the mux and process implementations. Can we continue to share them?

Reply:
One of the key difference between them is that for `mux_client_impl::just_connect`, the stdout and stderr is written into a file instead of as a pipe, since `mux_client_impl::{Session, Command, RemoteChild}` doesn't need to parse stdout/stderr of ssh multiplex server to get the error, however it might still need to read from them when creating the ssh multiplex server.

Reply:
It can be extracted though if I also change `process_impl::Session` to use read the log file instead.

Reply:
I have refactor `{mux_client_impl, process_impl}::just_connect` and extract out function `build_tempdir` and `launch_mux_master` and put them in `SessionBuilder`.

This has dramatically simplify the code and enables me to remove all `pub(crate)`s I added to `builder.rs`.